pkg/chaosctl/common/exec.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	"google.golang.org/grpc/grpclog"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/remotecommand"
+	kubectlscheme "k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+// Exec executes certain command and returns the result
+// runtime-controller only support CRUD， use client-go client
+func Exec(ctx context.Context, pod v1.Pod, daemon v1.Pod, cmd string, c *kubernetes.Clientset) (string, error) {
```

Comment:
I don't think we should try to run the command in the target pod firstly. Running all commands from the daemon is easier to understand.

And the same command could have different behavior (especially for busybox districution like alpine), so running commands in `chaos-daemon` directly could be much more reliable.

Reply:
Yes you're right. 
If I take it right, as @STRRL said, we need to change the order of these two methods: firstly go through chaos-daemon, if that's not working then directly execute in target pod. Or we just remove the backup plan, so the errors from going through `chaos-damon` would not be covered and we could know it and fix it.

Reply:
I think we don't need the "backup plan".

All commands should already be considered about "use chaos-daemon" or "run directly" before it executes.

Reply:
@STRRL Sure that would be better. Thanks!

It just which kind of command should be "run directly". Or it's just for future consideration, that for now everything could go through chaos-daemon

Reply:
It's just for future consideration. XD

Reply:
Thanks again for the refactor suggestion! 
Since currently we need to execute commands in both chaos-mesh components (mostly daemon but one for controller-manager) and target pods, so the commands executed in chaos-mesh components could directly use the first direct one, which is easy to reuse when we need 😆